advanced-creation.md

```diff
@@ -0,0 +1,163 @@
+# Advanced creation
+
+> Make calling REST APIs easier by creating niche-specific `got` instances.
+
+#### got.extend([options])
```

Comment:
`got.extend()` should stay in the readme, it was only `got.create()` that supposed to be moved out.

Reply:
Oh, right.